src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2619,9 +2677,102 @@ public enum SunMoonPromos implements LogicCardInfo {
+            top4.showToMe("The top 4 cards of your deck.")
+            top4.select(max:1,"Choose a card to put in your hand.").moveTo(my.hand)
+            shuffleDeck()
+            }
+          }
+        };
+      case CHARIZARD_GX_SM195:
+        return evolution (this, from:"Charmeleon", hp:HP250, type:R, retreatCost:4) {
+          weakness W
+          move "Raging Destruction", {
+            text "Discard the top 8 cards of your deck. If any of those cards are [R] Energy cards, attach them to this Pokémon."
+            energyCost R
+            attackRequirement {
+              assert my.deck : "Your deck is empty."
+            }
+            onAttack {
+              def top8 = my.deck.subList(0,8)
+              def list = top8.filterByType(BASIC_ENERGY).filterByEnergyType(R)
+              def num = list.size()
+              top8.showToMe("The top 8 cards of your deck.")
+              list.showToMe("These [R] Energies will be attached.")
+              if (num){
+                list.each{attachEnergy(self, it)}
+                }
+              }
+              my.deck.subList(0,8-num).discard()
+            }
+          }
+          move "Steam Artillery", {
+            text "200 damage."
+            energyCost R, R, C, C, C
+            onAttack {
+              damage 200
+            }
+          }
+          move "Dreadful Flames GX", {
+            text "250 damage. Discard an Energy from each of your opponent's Pokémon. (You can't use more than 1 GX attack in a game.)"
+            energyCost R, R, C, C, C
+            attackRequirement {
+              gxCheck()
+            }
+            onAttack {
+              gxPerform()
+              damage 250
+              opp.all.each{
+                if (it.cards.filterByType(ENERGY)) {
+                  it.cards.filterByType(ENERGY).select("Discard an Energy.").discard()
+            }
+          }
+        };
+      case MEWTWO_GX_SM196:
+        return basic (this, hp:HP190, type:P, retreatCost:2) {
+        weakness P
+        move "Telekinesis", {
+          text "This attack does 50 damage to 1 of your opponent's Pokémon. This damage isn't affected by Weakness or Resistance."
+          energyCost P, C
+          onAttack {
+            noWrDamage 50, opp.all.select()
+          }
+        }
+        move "Reigning Pulse", {
+          text "120 damage. Your opponent’s Active Pokémon is now Confused."
+          energyCost P, P, C, C
+          onAttack {
+            damage 120
+            apply CONFUSED
+          }
+        }
+        move "Psychic Nova GX", {
+          text "180 damage. Prevent all damage done to this Pokémon by attacks during your opponent's next turn. (You can't use more than 1 GX attack in a game.)"
+          energyCost P, P, C, C
+          attackRequirement {
+            gxCheck()
+          }
+          onAttack {
+            gxPerform()
+            damage 180
+            preventAllEffectsNextTurn()
+            }
```

Comment:
This appears to be an extra bracket

Reply:
This is the ending bracket to the `onAttack` field for Psychic Nova GX.

Reply:
Oh sorry about that, the indenting on github is throwing me off 😭